### PR TITLE
fixed bugs in the keep population flag.

### PR DESCRIPTION
**Pull request by SwiftySalmon**
- filter out where gender not 1 or 2
- deal with negative ages where dob is after mid point of year
- previously filtered cases where dod was before midpoint, but was also accidentally removing cases with missing death dates! Now fixed.
- added filter for nsu only once calculations are done

### CODE REVIEW

**Review by lizihao-anu — DISMISSED** [review]
Thanks for the fixing

File: R/add_keep_population_flag.R [hunk 1]
```diff
@@ -137,7 +139,7 @@ add_age_group <- function(data, age_var_name) {
   data <- data %>%
     dplyr::mutate(
       age_group = dplyr::case_when(
-        {{ age_var_name }} >= 0 & {{ age_var_name }} <= 4 ~ "0-4",
+        {{ age_var_name }} >= -1 & {{ age_var_name }} <= 4 ~ "0-4",
```
Comment (rchlv) [inline R/add_keep_population_flag.R:142]: Thanks for figuring out this issue, well done!
For age, do we sometimes get people with age -1?

Comment (SwiftySalmon) [reply R/add_keep_population_flag.R:142]: I think it's due to age being calculated at the midpoint of the year. So if someone is born in September their age on 30th June is -1. Very fun for us.

File: R/add_keep_population_flag.R [hunk 2]
```diff
@@ -70,17 +71,18 @@ add_keep_population_flag <- function(individual_file, year) {
...
-      # anyone who died 5 years before the file shouldn't be in it anyway...
-      dplyr::filter(death_date > mid_year | nsu != 0) %>%
+      dplyr::filter(death_date > mid_year | is.na(death_date) | nsu != 0) %>%
       # Calculate the populations of the whole SLF and of the NSU.
       dplyr::group_by(locality, age_group, gender) %>%
       dplyr::mutate(
         nsu_population = sum(nsu),
         total_source_population = dplyr::n()
       ) %>%
+      dplyr::filter(nsu == 1) %>%
```
Comment (Jennit07) [inline R/add_keep_population_flag.R:85]: This was a good spot here! I've checked SPSS and this matches up. I think im happy to approve this now!

**Review by Jennit07 — APPROVED** [review]

**Pull request merged**